astacus/node/api.py

```diff
@@ -115,11 +130,26 @@ def upload_result(*, op_id: int, n: Node = Depends()):
     return op.result
 
 
+@router.post("/delta/upload")
+def delta_upload(req: ipc.SnapshotUploadRequestV20221129, n: Node = Depends()):
+    if not n.state.is_locked:
+        raise HTTPException(status_code=409, detail="Not locked")
+    snapshotter = n.get_delta_snapshotter()
+    assert snapshotter
+    return UploadOp(n=n, op_id=n.allocate_op_id(), stats=n.stats, req=req).start(snapshotter)
+
+
+@router.get("/delta/upload/{op_id}")
+def delta_upload_result(*, op_id: int, n: Node = Depends()):
+    op, _ = n.get_op_and_op_info(op_id=op_id, op_name=OpName.upload)
+    return op.result
+
+
 @router.post("/download")
 def download(req: ipc.SnapshotDownloadRequest, n: Node = Depends()):
     if not n.state.is_locked:
         raise HTTPException(status_code=409, detail="Not locked")
-    return DownloadOp(n=n, op_id=n.allocate_op_id(), stats=n.stats, req=req).start()
+    return DownloadOp(n=n, op_id=n.allocate_op_id(), stats=n.stats, req=req).start(n.get_or_create_snapshotter)
```

Comment:
This looks a bit too hard to follow with the init parameters but also a callback passed later, the init takes a `req` and then the callback will be called with things from `req` that was passed just before that. I think it's worth simplifying and just passing a configured snapshotter from the API to the Op, without callbacks.